src/pip/_internal/commands/uninstall.py

```diff
@@ -59,6 +61,17 @@ def run(self, options, args):
 
         reqs_to_uninstall = {}
         for name in args:
+            if is_archive(name):
+                raise InstallationError(
+                    'You can not pass an archive as a parameter '
+                    'to "pip uninstall" (see "pip help {self.name}")'.format(**locals())
```

Comment:
why use `self.name`?

Reply:
I prefer using the `self.name` instead of hard coding the `uninstall` command. I updated the error messages accordingly